core/store/orm/locking_strategies_test.go

```diff
@@ -85,3 +88,69 @@ func TestPostgresLockingStrategy_Lock(t *testing.T) {
 	require.NoError(t, ls2.Lock(delay), "should get exclusive lock")
 	require.NoError(t, ls2.Unlock())
 }
+
+func TestPostgresLockingStrategy_WhenLostIsReacquired(t *testing.T) {
+	t.Parallel()
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	if store.Config.DatabaseURL() == "" {
+		t.Skip("No postgres DatabaseURL set.")
+	}
+
+	err := store.ORM.LockingStrategy().Unlock()
+	require.NoError(t, err)
+
+	var jobs []models.JobSpec
+	err = store.ORM.DB.Find(&jobs).Error
+	require.NoError(t, err)
```

Comment:
Suggestion: This implies that the lock is reacquired because we know the implementation code. Let's make that explicit with a subsequent check. Have a `lock2` try to acquire the lock, and fail.